modules/collections/client/controllers/collections.controllers.js

```diff
@@ -159,145 +164,210 @@ collectionModules.controller('CollectionEditCtrl',
-          break;
+      }
 
-        case 'Permit':
-          $scope.collection.parentType = 'Authorizations';
-          break;
-
-        case 'Inspection Report':
-        case 'Order':
-          $scope.collection.parentType = 'Compliance and Enforcement';
-          break;
+      // Cancel any unsaved changes on the page.
+      $scope.cancel = function() {
+        if ($scope.collectionForm.$dirty || $scope.validationFlags.docsChanged) {
+          ConfirmService.confirmDialog({
+            titleText: "Warning: Unsaved Changes",
+            confirmText: "Are you sure you want to leave this page?",
+            okText: 'Yes',
+            cancelText: 'No',
+            onOk: function() {
+              $scope.goToList();
+            }
+          });
+        } else {
+          $scope.goToList();
+        }
+      }
 
-        case 'Annual Report':
-        case 'Management Plan':
-        case 'Dam Safety Inspection':
-        case 'Letter of Assurance':
-          $scope.collection.parentType = 'Other';
-          break;
+      // Delete a collection.
+      $scope.delete = function() {
+        if ($scope.collection.isPublished) {
+          AlertService.error("Error: published collections cannot be deleted", 4000, true);
+          return;
         }
-        CollectionModel.add($scope.collection)
-          .then(function(/* model */) {
-            // var newparam = 5003;
-            AlertService.success('"'+ $scope.collection.displayName +'" was successfully created.', 4000, true);
-            $state.transitionTo('p.collection.edit', { projectid: project.code, collectionId: collection._id }, {
-              reload: true, inherit: false, notify: true
+        var modalView = $uibModal.open({
+          animation: true,
+          templateUrl: 'modules/utils/client/views/partials/modal-confirm-delete.html',
+          controller: function($scope, $state, $uibModalInstance) {
+            var self = this;
+            self.dialogTitle = "Delete Collection";
+            self.name = $scope.collection.displayName;
+            self.ok = function() {
+              $uibModalInstance.close($scope.collection);
+            };
+            self.cancel = function() {
+              $uibModalInstance.dismiss('cancel');
+            };
+          },
+          controllerAs: 'self',
+          scope: $scope,
+          size: 'md'
+        });
+        modalView.result.then(function() {
+          CollectionModel.deleteId($scope.collection._id)
+            .then(function() {
+              // Collection deleted. Show confirmation message and return to list.
+              AlertService.success('"'+ $scope.collection.displayName +'" was deleted successfully.', 4000, true);
+              $scope.goToList();
+            })
+            .catch(function() {
+              // User cancels delete, or error is thrown.
+              AlertService.error('"'+ $scope.collection.displayName +'" was not deleted.', 4000, true);
```

Comment:
Same comment again: this is a third style of error message.  Seems like we should unify what our messaging is.
Maybe this is for another story though.

Reply:
I've decided to go with this particular wording, and changed the rest to match it. Seems concise enough.